docs(logs): add docs for posthog.logger (#16461)
PostHog/posthog.com: contents/docs/logs/installation/javascript.mdx

---
title: JavaScript web logs installation
platformLogo: javascript
showStepsToc: true
---

import { Steps, Step } from 'components/Docs/Steps'
import LogsNextSteps from './_snippets/logs-next-steps.mdx'

> The PostHog JavaScript web SDK ships with first-class support for Logs. Send structured log records with the `posthog.logger` API, or opt in to autocapture of `console.*` calls – or both.

Reviewer: ```suggestion
The PostHog JavaScript web SDK ships with first-class support for Logs. Send structured log records with the `posthog.logger` API or opt in to autocapture of `console.*` calls.
```

I feel like it's self-evident that you do both.

Reviewer: A nit, up to you :D

> **Minimum version:** `posthog-js@1.368.0` or later. Run `npm update posthog-js` (or your package manager's equivalent) to update.

<Steps>

<Step title="Install posthog-js" badge="required">

> Install and initialize `posthog-js` if you haven't already. See the [web SDK install guide](/docs/libraries/js) for setup details.

Reviewer: There should be snippets you can reuse here. We should keep all info on this single page and use snippet to deduplicate.

Reviewer: (please fix this quickly before you merge)

| `serviceVersion` | Release version, e.g. `'1.2.3'`. Maps to `service.version`. |
| `resourceAttributes` | Additional OpenTelemetry resource attributes, e.g. `{ 'host.name': 'web-01' }`. |
| `flushIntervalMs` | How often batched log records are flushed. Defaults to `3000`. |
| `maxBufferSize` | Buffer size before forcing a flush. Defaults to `100`. |
| `maxLogsPerInterval` | Rate limit per flush window. Excess records are dropped with a single warning. Defaults to `1000`. |

See the [web SDK config reference](/docs/libraries/js/config#configuring-logs) for the complete list.

</Step>

<Step title="Send structured logs with posthog.logger" badge="recommended">

`posthog.logger` is the declarative path. Each call produces an OTLP log record with a severity level and typed attributes you can filter and query on in the Logs product.

```js-web
posthog.logger.info('checkout completed', { order_id: 'ord_789', total_cents: 4200 })
posthog.logger.warn('payment retry', { attempt: 2, gateway: 'stripe' })
posthog.logger.error('payment failed', { error_code: 'E001', order_id: 'ord_789' })
```

Available severity levels: `trace`, `debug`, `info`, `warn`, `error`, and `fatal`.

> Logger calls don't write to the browser console. They go directly to PostHog as structured records, so you can query by attribute (e.g. `total_cents > 1000`) rather than string-searching log output, and nothing extra appears in the user's browser console.

Reviewer: ```suggestion
Logger calls don't write to the browser console, they go directly to PostHog as structured records. You can query logs by attribute, such as `total_cents > 1000`, instead of string-searching log output. Nothing extra appears in the user's browser console.
```

```js-web
posthog.captureLog({
  body: 'checkout completed',
  level: 'info',
  attributes: { order_id: 'ord_789', tags: ['checkout', 'v2'] },
  trace_id: '0af7651916cd43dd8448eb211c80319c', // optional, 32 hex chars
  span_id: 'b7ad6b7169203331',                   // optional, 16 hex chars
})
```

Attribute values can be strings, numbers, booleans, arrays, or plain objects. `null` and `undefined` values are dropped.

</Step>

<Step title="Autocapture console logs" badge="optional">

To forward `console.log`, `console.warn`, `console.error`, and friends without changing your code, set `logs.captureConsoleLogs: true`.

```js-web
posthog.init('<ph_project_token>', {
  api_host: '<ph_client_api_host>',
  defaults: '<ph_posthog_js_defaults>',
  logs: {
    captureConsoleLogs: true,
    serviceName: 'checkout-web',
  },
})
```

> **Warning:** Overriding a system library like `console` has a large blast radius. Everything your app or its dependencies write to the console gets forwarded to PostHog, including:
>
> - Debug output referencing passwords, tokens, API keys, or session identifiers
> - Stack traces containing user input
> - Third-party library chatter you don't control
>

> > This can create compliance exposure and is painful to clean up after the fact. Before turning it on:

Reviewer: ```suggestion
> This can breach data privacy compliance and is difficult to clean up. Before turning it on:
```
